django/db/models/fields/related_lookups.py

```diff
@@ -83,6 +83,29 @@ def as_sql(self, compiler, connection):
         else:
             return super(RelatedIn, self).as_sql(compiler, connection)
 
+    def __getstate__(self):
+        from django.db.models.query import QuerySet
+        obj_dict = self.__dict__.copy()
+        del obj_dict['rhs']
+        if isinstance(self.rhs, QuerySet):
+            # Pickle query fron queryset only
+            return (self.rhs.model, self.rhs.query, obj_dict)
+        else:
+            # Pickle RHS as is
+            return (None, self.rhs, obj_dict)
+
+    def __setstate__(self, state):
+        model, rhs, obj_dict = state
+        if model is None:
+            # Unpickle full RHS
+            self.rhs = rhs
+        else:
+            # Unpickle RHS as query
+            queryset = model.objects.none()
+            queryset.query = rhs
+            self.rhs = queryset
```

Comment:
Actually it doesn't work as expected. This is the original inner query:
`WHERE "queryset_pickle_event"."group_id" IN (SELECT "queryset_pickle_group"."id" FROM "queryset_pickle_group")`

And this is unpickled inner query:
`WHERE "queryset_pickle_event"."group_id" IN (SELECT "queryset_pickle_group"."id" AS Col1, "queryset_pickle_group"."name" AS Col2 FROM "queryset_pickle_group")
`

For some reason query itself doesn't preserve queried fields but it tries to query by (all) fields in a model.

Trying to evaluate unpickled query results to `OperationalError: only a single result allowed for a SELECT that is part of an expression` exception,